pkg/syncclient/sync_client.go

```diff
@@ -165,6 +176,26 @@ func (s *SyncerClient) Start(cxt context.Context) error {
 	return nil
 }
 
+// SupportsNodeResourceUpdates waits for the Typha server to send a hello and returns true if
+// the server supports node resource updates. If the given timeout is reached, an error is returned.
+func (s *SyncerClient) SupportsNodeResourceUpdates(timeout time.Duration) (bool, error) {
+	// If we've already gotten a MsgServerHello just return with the value
+	if s.handshakeStatus.complete {
```

Comment:
I don't think the comment here is quite right.  If I'm reading the code correctly, the only way for `s.handshakeStatus.complete` to be true is if `SupportsNodeResourceUpdates` has already been called before.  So the comment should read "If a previous call has already marked the handshake as complete, just ..."